src/target/riscv/riscv-013.c

```diff
@@ -387,52 +387,47 @@ static unsigned int decode_dm(char *text, unsigned int address, unsigned int dat
 	return 0;
 }
 
-static unsigned int decode_dmi(struct target *target, char *text, unsigned int address,
-		unsigned int data)
-{
-	dm013_info_t *dm = get_dm(target);
-	if (!dm) {
-		if (text)
-			text[0] = '\0';
-		return 0;
-	}
-	return decode_dm(text, address - dm->base, data);
-}
-
-static void dump_field(struct target *target, int idle, const struct scan_field *field, bool discard_in)
+void riscv_dump_field(const struct target *target, int idle, const struct scan_field *field, bool discard_in)
```

Comment:
While making changes here, I'd recommend to rename the function to `riscv_decode_jtag_scan()`, `riscv_log_jtag_scan()` or similar. The word `field` is probably not be very clear to readers.